src/main/java/fr/pilato/elasticsearch/crawler/fs/FsCrawlerImpl.java

```diff
@@ -568,33 +575,40 @@ private void indexFile(FileAbstractModel fileAbstractModel, ScanStatistic stats,
                         // https://github.com/dadoonet/fscrawler/issues/5 : Support JSon files
-                        esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
-                                fsSettings.getElasticsearch().getType(),
-                                generateIdFromFilename(filename, dirname),
-                                read(inputStream));
-                        return;
-                    } else if (fsSettings.getFs().isXmlSupport()) {
+                        if (fsSettings.getFs().isAddAsInnerObject()) {
+                            esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
+                                    fsSettings.getElasticsearch().getType(),
+                                    generateIdFromFilename(filename, dirname),
+                                    read(inputStream));
+                            return;
+                        } else {
+                                // https://github.com/dadoonet/fscrawler/issues/5 : Support JSon files
+                                doc.setObject(DocParser.fromJsonToMap(read(inputStream)));
+                        }
+                    } else if (fsSettings.getFs().isXmlSupport() && fileExtension.equals("xml")) {
                         // https://github.com/dadoonet/fscrawler/issues/185 : Support Xml files
-                        esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
-                                fsSettings.getElasticsearch().getType(),
-                                generateIdFromFilename(filename, dirname),
-                                XmlDocParser.generate(inputStream));
-                        return;
+                        if (fsSettings.getFs().isAddAsInnerObject()) {
+                            esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
+                                    fsSettings.getElasticsearch().getType(),
+                                    generateIdFromFilename(filename, dirname),
+                                    XmlDocParser.generate(inputStream));
+                            return;
+                        } else {
+                                // https://github.com/dadoonet/fscrawler/issues/185 : Support Xml files
+                                doc.setObject(XmlDocParser.generateMap(inputStream));
+                        }
                     } else {
-                        // Extracting content with Tika
                         generate(fsSettings, inputStream, filename, doc, messageDigest, filesize);
                     }
                 }
-
-                // We index
                 esIndex(esClientManager.bulkProcessor(), fsSettings.getElasticsearch().getIndex(),
                         fsSettings.getElasticsearch().getType(),
-                        generateIdFromFilename(filename, dirname),
+                        docId,
```

Comment:
`generateIdFromFilename` does already what is needed. This change is not needed anymore I think.